### PR TITLE
Fix a few issues with forensics layer

### PR DESCRIPTION
**Pull request by silasdavis**
Signed-off-by: Silas Davis <silas@monax.io>

### CODE REVIEW

File: execution/evm/vm.go [hunk 1]
```diff
@@ -134,13 +134,18 @@ func (vm *VM) fireCallEvent(eventSink EventSink, callType exec.CallType, errProv
 	errSink.PushError(eventErr)
 }
 
+func (vm *VM) Call(callState Interface, eventSink EventSink, caller, callee crypto.Address, code,
+	input []byte, value uint64, gas *uint64) (output []byte, err errors.CodedError) {
+	return vm.CallChipsBLah(callState, eventSink, caller, callee, code, input, value, gas)
```
Comment (seanyoung) [inline execution/evm/vm.go]: Eh?

File: execution/evm/vm.go [hunk 2]
```diff
@@ -134,12 +134,9 @@ func (vm *VM) fireCallEvent(eventSink EventSink, callType exec.CallType, errProv
...
 }
 
-// CONTRACT state is aware of caller and callee, so we can just mutate them.
-// CONTRACT code and input are not mutated.
-// CONTRACT returned 'ret' is a new compact slice.
-// value: To be transferred from caller to callee. Refunded upon errors.CodedError.
-// gas:   Available gas. No refunds for gas.
-// code: May be nil, since the CALL opcode may be used to send value from contracts to accounts
+// Initiate an EVM call against the provided state pushing events to eventSink. code should contain the EVM bytecode,
+// input the CallData (readable by CALLDATALOAD), value the amount of native token to transfer with the call
```
Comment (seanyoung) [inline execution/evm/vm.go]: This is not entirely correct. In the deploy code path the constructor arguments are concatenated with the code, and not stored in the calldataload. This is just another EVM weirdism.

File: forensics/replay.go [hunk 3]
```diff
@@ -131,70 +131,64 @@ func (re *Replay) Block(height uint64) (*ReplayCapture, error) {
...
 		if err != nil {
-			return nil, err
-		}
-		// block.AppHash is hash after txs from previous block have been applied - it's the state we want to load on top
-		// of which we will reapply this block txs
-		st, err = re.State(startHeight - 1)
-		if err != nil {
-			return nil, err
-		}
-	} else {
```
Comment (gregdhill) [inline forensics/replay.go]: @silasdavis isn't this required for a non-restored chain?

Comment (silasdavis) [reply forensics/replay.go]: @gregdhill no, when I wrote it I thought it was, when we pull state at height 0, we get state version 1, which is the initial commit containing the genesis state as originally laid down. So this is better than us recreating it here and doesn't require a special code path.

**Pull request merged**